src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -0,0 +1,249 @@
+package com.android.volley.toolbox;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.VisibleForTesting;
+import com.android.volley.AsyncCache;
+import com.android.volley.Header;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressLint("NewApi")
+public class DiskBasedAsyncCache extends AsyncCache {
```

Comment:
As for `AsyncCache`, make sure to add javadoc to describe what the class does and how to use it.